src/library/scala/collection/immutable/Range.scala

```diff
@@ -518,4 +518,17 @@ private class RangeIterator(
     _next = value + step
     value
   }
+
+  override def drop(n: Int): Iterator[Int] = {
+    val longPos = _next.toLong + step * n
+    if (step > 0) {
+      _next = Math.min(lastElement, longPos).toInt
```

Comment:
`_next = longPos.toInt` should be sufficient. `_next` is not used when `_hasNext == false` so there's no need to clamp it back to `lastElement` (which would be wrong anyway because we're already past the last element)